src/lava/lib/optimization/solvers/generic/solution_finder/process.py

```diff
@@ -15,23 +15,29 @@ def __init__(
         cost_diagonal,
         cost_coefficients,
         constraints,
+        backend,
         hyperparameters,
         discrete_var_shape,
         continuous_var_shape,
+        problem,
         name: ty.Optional[str] = None,
         log_config: ty.Optional[LogConfig] = None,
     ):
         super().__init__(
             cost_diagonal=cost_diagonal,
             cost_coefficients=cost_coefficients,
             constraints=constraints,
+            backend=backend,
             hyperparameters=hyperparameters,
             discrete_var_shape=discrete_var_shape,
             continuous_var_shape=continuous_var_shape,
+            problem=problem,
             name=name,
             log_config=log_config,
         )
-        self.variables_assignment = Var(shape=discrete_var_shape, init=(1,))
+        self.variables_assignment = Var(
+            shape=discrete_var_shape or continuous_var_shape, init=(1,)
```

Comment:
Hm... For the future, could we use discrete_var_shape + continuous_var_shape here (of course accounting for None values)?

Reply:
We could in the future